lib/domain/usecases/products/find_products_by_filter_use_case.dart

```diff
@@ -1,6 +1,59 @@
+import 'dart:collection';
+
+import 'package:openflutterecommerce/data/abstract/model/filter_rules.dart';
+import 'package:openflutterecommerce/data/abstract/model/product.dart';
+import 'package:openflutterecommerce/data/abstract/model/product_attribute.dart';
+import 'package:openflutterecommerce/data/abstract/product_repository.dart';
 import 'package:openflutterecommerce/domain/usecases/base_use_case.dart';
 import 'package:openflutterecommerce/domain/usecases/products/products_by_filter_params.dart';
 import 'package:openflutterecommerce/domain/usecases/products/products_by_filter_result.dart';
+import 'package:openflutterecommerce/locator.dart';
 
 abstract class FindProductsByFilterUseCase
-    implements BaseUseCase<ProductsByFilterResult, ProductsByFilterParams> {}
\ No newline at end of file
+    implements BaseUseCase<ProductsByFilterResult, ProductsByFilterParams> {}
+
+    
+class FindProductsByFilterUseCaseImpl implements FindProductsByFilterUseCase {
+
+  @override
+  Future<ProductsByFilterResult> execute(
+      ProductsByFilterParams params) async {
+    try {
+      var products = await _findProductsByFilter(params);
+
+      if (products != null && products.isNotEmpty) {
+        HashMap<ProductAttribute, List<String>> result = HashMap();
+        products.forEach((product) => 
+          product.selectableAttributes != null ?
+            result.addAll({for (var attribute in product.selectableAttributes) attribute: []})
```

Comment:
what does it do?

Reply:
That should generate the selectable attributes for the future filter screen in list of products.